src/widget/wtrackmenu.cpp

```diff
@@ -963,10 +967,10 @@ void WTrackMenu::slotOpenInFileBrowser() {
     mixxx::DesktopHelper::openInFileBrowser(locations);
 }
 
-void WTrackMenu::slotFindOnSoundcloud(const Track& track) {
-    const auto artistName = track.getArtist();
-
-    QDesktopServices::openUrl(QUrl("https://soundcloud.com/search/people?q="));
+void WTrackMenu::slotFindOnSoundcloud() {
+    const TrackPointer pTrack = getFirstTrackPointer();
+    QString artistName = pTrack->getArtist();
```

Comment:
This could be empty. Use album artist as fallback?
If that is also empty, ask the user? Or just open the soundcloud search and let the user type there?

Reply:
Yeah, album artist and asking the user might be good option. I can implement that, If it's okay.

Right now it opens SoundCloud with a blank page.

Reply:
I think we need to % encode the artist name. 

Opening SoundCloud with a blank page sounds good enough for me, to realize that the artist is empty. 

What exactly do we want to ask the user? What will be the alternative? We may gray out the menu entry or open the track property dialog to add an artist.

Reply:
> I think we need to % encode the artist name.

Oh I just realized that QUrl is smart enough to do it automatically.

Reply:
> What exactly do we want to ask the user? What will be the alternative? We may gray out the menu entry or open the track property dialog to add an artist.

Graying out the menu action is not helpful IMO. Initially I thought of a simple textbox dialog asking for the track artist, apply that to the track, then attempting to open soundcloud again. But that's too cumbersome UX-wise and to program for this first step.

Since an empty soundcloud query would not solve the missing tag issue we could show a dialog
> This track has no artist or album artist tag.
Try again after filling the tag(s) in the track properties dialog or inline in the tracks table.
[ Open track properties] [ Okay ]

Reply:
Keep it simple without requiring any additional user input, see the "Search related Tracks" feature. If I need to type I could open a browser myself.

Mixxx should not become a launcher for external applications. Opening the file explorer is a notable but useful exception. If a feature can't be integrated properly into Mixxx it should not become part of Mixxx. The feature as currently planned is already an edge case IMHO.

Reply:
I have looked at "Search Related Tracks" feature it is so nice. When there is no artist info. It doesn't show the find artist row. This feature can be like this too, so no user input needed and also can be extended, such as find the artist, albumartist, track on lastfm discogs or musicbrainz in a single menu such as "Search on web".

Opening the file browser is a must and it is very useful. Launching the web browser can be useful too. I think for some users It can be helpful to find the artists on different services. If they are looking for information on the web a lot about the artist, such as latest releases, albums etc. It can save their time.